src/index.js

```diff
@@ -81,14 +82,25 @@ export default class ScrollPlugin extends Plugin {
         window.history.scrollRestoration = 'auto';
     }
 
+    getOffset = () => {
+        switch (typeof this.options.offset) {
+            case 'number':
+                return this.options.offset;
+            case 'function':
+                return this.options.offset();
+            default:
+                return Number(this.options.offset);
+        }
+    }
+
     onSamePage = () => {
         this.swup.scrollTo(0);
     }
 
     onSamePageWithHash = event => {
         const link = event.delegateTarget;
         const element = document.querySelector(link.hash);
-        const top = element.getBoundingClientRect().top + window.pageYOffset;
+        const top = element.getBoundingClientRect().top + window.pageYOffset - this.getOffset();
```

Comment:
Since we can use a function that can return a dynamic offset, would it make sense to pass the element to it as well, since it could hold some useful information? I don't have any specific use case tho, just thinking outloud.

Reply:
Great idea, could be useful. I'll make the necessary changes.